src/api.ts

```diff
@@ -73,3 +73,39 @@ function validateProject(project: Project): boolean {
 
   return true;
 }
+
+function addProject(devId: number, project: Project) {
+  const developer = developers.find((dev) => dev.id === devId);
+
+  if (!developer) {
+    throw Error(`Error: Developer with id: ${devId} does not exist`);
+  }
+
+  if (validateProject(project)) {
+    throw Error("Error: Project is not valid");
+  }
+
+  developer.projects.push(project);
+}
+
+function listProjects(devId: number) {
+  const developer = developers.find((dev) => dev.id === devId);
+
+  if (!developer) {
+    throw Error(`Error: Developer with id: ${devId} does not exist`);
+  }
+
+  const projectsList = developer.projects.map(({ projectName }) => projectName);
+
+  return projectsList;
+}
+
+function countCompletedProjects(devId: number) {
+  const developer = developers.find((dev) => dev.id === devId);
+
+  if (!developer) {
+    throw Error(`Error: Developer with id: ${devId} does not exist`);
+  }
+
+  return developer.projects.length;
```

Comment:
is this assuming all the projects are complete?

Reply:
Yeah I just found it out, working on it.